### ISSUE TITLE
Server Error on clicking on the DocType Browser

### ISSUE THREAD

**Issue by charlesroper** [DanDiplo/Umbraco.GodMode#2]
Hi Dan,

I've just installed God Mode on a large site I've been upgrading. It's got tons of DocTypes and goes back to Umbraco 4, although I'm upgrading it from 6 at this stage.

The issue I am seeing is that it says "Server error: Contact administrator, see log for full details. An error occurred!" when clicking on the DocType Browser. The spinner stays on the screen and it never loads.

It says to look in the Umbraco trace log (App_Data > Logs), but I don't see anything in the current log related to this issue. If I clear it out and reproduce the error, the log is empty.

The console throws a 500 error.

Any ideas?

![2017-03-13_15h54_34](https://cloud.githubusercontent.com/assets/176/23862935/47adcb8e-0806-11e7-80fc-3cf59f544c10.png)

![2017-03-13_15h57_46](https://cloud.githubusercontent.com/assets/176/23862949/4e8c791e-0806-11e7-83aa-c07ff2fe3ca7.png)

**Comment by DanDiplo** [comment]
Hi Charles,

Hmmm, strange, it's not an error I've seen before. The red error banner is something Umbraco is generating - I would assume Umbraco would also log the exception at this point, but clearly not. I didn't do any of my own exception handling at this level as I thought Umbraco did. You could maybe try running your project in debug mode, it may yield something more.

Most of the code for this method is basically calling the core Umbraco content Services that queries the the data and then maps them to models (which are returned to the Angular view). But without any error stack trace it's hard to know what is at fault. I can point you to the code being called at this point: 

https://github.com/DanDiplo/Umbraco.GodMode/blob/master/Diplo.GodMode/Services/UmbracoDataService.cs#L37

All I can really suggest is grabbing the source code and adding some break points and/or logging calls. It's not an error anyone has flagged before, or I've seen before, so I'm guessing that maybe something in the upgrades has resulted in some data going missing?!? If you can find any specific errors I'll be able to help more.

**Comment by charlesroper** [comment]
Not sure if this is helpful or not, but when running in debug mode, I get the following when I hit the page:

Received an error from the server
An error occured

The 'ObjectContent`1' type failed to serialize the response body for content type 'application/json; charset=utf-8'.

EXCEPTION DETAILS

System.InvalidOperationException: The 'ObjectContent`1' type failed to serialize the response body for content type 'application/json; charset=utf-8'.
STACKTRACE

INNER EXCEPTION

```
System.NullReferenceException: Object reference not set to an instance of an object.
at Diplo.GodMode.Services.UmbracoDataService.<>c__DisplayClass11.<GetContentTypeMap>b__3(ITemplate x)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeList(JsonWriter writer, IEnumerable values, JsonArrayContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeValue(JsonWriter writer, Object value, JsonContract valueContract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeValue(JsonWriter writer, Object value, JsonContract valueContract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeList(JsonWriter writer, IEnumerable values, JsonArrayContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeValue(JsonWriter writer, Object value, JsonContract valueContract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.Serialize(JsonWriter jsonWriter, Object value, Type objectType)
   at Newtonsoft.Json.JsonSerializer.SerializeInternal(JsonWriter jsonWriter, Object value, Type objectType)
   at System.Net.Http.Formatting.BaseJsonMediaTypeFormatter.WriteToStream(Type type, Object value, Stream writeStream, Encoding effectiveEncoding)
   at System.Net.Http.Formatting.JsonMediaTypeFormatter.WriteToStream(Type type, Object value, Stream writeStream, Encoding effectiveEncoding)
   at System.Net.Http.Formatting.BaseJsonMediaTypeFormatter.WriteToStream(Type type, Object value, Stream writeStream, HttpContent content)
   at System.Net.Http.Formatting.BaseJsonMediaTypeFormatter.WriteToStreamAsync(Type type, Object value, Stream writeStream, HttpContent content, TransportContext transportContext, CancellationToken cancellationToken)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.WebHost.HttpControllerHandler.<WriteBufferedResponseContentAsync>d__1b.MoveNext()
```

**Comment by DanDiplo** [comment]
Hmmm, it seems it's not able to serialise the models, for some reason. I've seen this on StackOverflow - http://stackoverflow.com/questions/12641386/failed-to-serialize-the-response-in-web-api - so there may be something to experiement with.  But it's going to be hard to fix unless I can replicate - AFAIK you are the only person to encounter this :)

**Comment by DanDiplo** [comment]
@charlesroper I've been able to replicate this - just required a few more null checks (I think Umbraco API slightly changed somewhere) - and have fixed it in latest commit.

I've attached a DLL which I hope fixes it - if you get the chance, give it a try and let me know how it works. If it fixes it, I'll release an update.

[Diplo.GodMode.dll.zip](https://github.com/DanDiplo/Umbraco.GodMode/files/910718/Diplo.GodMode.dll.zip)
